includes/Modules/Analytics_4/Report/Request.php

```diff
@@ -380,4 +388,123 @@ protected function parse_dimension_filter( $dimension_name, $dimension_value ) {
 		return $filter_expression;
 	}
 
+	/**
+	 * Parses metric filters and returns a filter expression that should be added to the report request.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Data_Request $data Data request object.
+	 * @return Google_Service_AnalyticsData_FilterExpression The filter expression to use with the report request.
+	 */
+	protected function parse_metric_filters( Data_Request $data ) {
+		$expressions = array();
+
+		if ( is_array( $data['metricFilters'] ) ) {
+			foreach ( $data['metricFilters'] as $key => $value ) {
+				$expressions[] = $this->parse_metric_filter( $key, $value );
+			}
+		}
+
+		if ( ! empty( $expressions ) ) {
+			$filter_expression_list = new Google_Service_AnalyticsData_FilterExpressionList();
+			$filter_expression_list->setExpressions( array_filter( $expressions ) );
+
+			$metric_filters = new Google_Service_AnalyticsData_FilterExpression();
+			$metric_filters->setAndGroup( $filter_expression_list );
+
+			return $metric_filters;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Parses and returns a single metric filter.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $metric_name The metric name.
+	 * @param mixed  $metric_value The metric filter settings.
+	 * @return Google_Service_AnalyticsData_FilterExpression The filter expression instance.
+	 */
+	protected function parse_metric_filter( $metric_name, $metric_value ) {
+		// Use the numeric filter type by default.
+		$filter_type = 'numericFilter';
+		if ( isset( $metric_value['filterType'] ) ) {
+			// If there is the filterType property, use the explicit filter type then.
+			$filter_type = $metric_value['filterType'];
+		}
+
+		if ( 'numericFilter' === $filter_type ) {
+			if ( ! isset( $metric_value['operation'] ) && ! isset( $metric_value['value'] ) ) {
+				return null;
```

Comment:
We might want to return an error here and elsewhere that the required parameters are not provided... However, the scope for this issue has already widened enough, and I don't think this is a critical aspect that needs addressing right now. Could you please create a followup issue to consider how we want to handle these cases? It can just be a brief summary, we can get into the detail later.

Reply:
Got it, although these also have a validations in `getReport` action, which checks for the required parameters to be present.

Reply:
Thanks @zutigrm, it's a good point. However, I do think our API should still behave as a "good API citizen", returning useful errors when something unexpected happens. This would both protect us from inadvertently calling it with the wrong parameters from our own code, and also ensure the endpoint behaves in a reasonable manner in the potential case where one of our users decides to use it in their own integration with the plugin.

Reply:
@techanvil Added - 7643

Reply:
Thanks @zutigrm! Btw, it's generally worth using the `#<issue_number>` syntax in order to create a link between the issues - I'll do it here and it should automatically create an entry on that issue linking back to this one. Here goes: #7643.

Reply:
@techanvil Thanks, will make sure to use it.